07.ls_object/file_info.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+class FileInfo
+  attr_reader :name, :path
+
+  TYPE_LIST = {
+    '01' => 'p',
+    '02' => 'c',
+    '04' => 'd',
+    '06' => 'b',
+    '10' => '-',
+    '12' => 'l',
+    '14' => 's'
+  }.freeze
+
+  PERMISSION_LIST = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  private_constant :TYPE_LIST, :PERMISSION_LIST
+
+  def initialize(path, name)
```

Comment:
```suggestion
  def initialize(file_path)
```

pathとnameに分けずに「ファイルのパス」をまるっと受け取るようにした方が自然な気がします。

Reply:
ファイルのパスを渡してインスタンスを作るよう変更しました。
インスタンス変数`@name`を不要に感じたので、メソッドに変更しました。https://github.com/SuzukaHori/ruby-practices/pull/12/commits/eb9284a973f56dddc43a8d7783501d2d34878f1c